service_container/calls.rst

```diff
@@ -66,8 +66,86 @@ To configure the container to call the ``setLogger`` method, use the ``calls`` k
     .. code-block:: php
 
         // config/services.php
-        use App\Service\MessageGenerator;
+        use App\Message\MessageGenerator;
         use Symfony\Component\DependencyInjection\Reference;
 
         $container->register(MessageGenerator::class)
             ->addMethodCall('setLogger', [new Reference('logger')]);
+
+
+.. versionadded:: 4.3
+
+    The ``immutable-setter`` injection was introduced in Symfony 4.3.
+
+When it comes to optional dependencies, the setter approach can be useful but
+what if you need to change the injected dependency without mutating the initial service?
```

Comment:
Alternative:
In order to provide immutable services, some classes implement immutable setters. Such setters return a new instance of the configured class instead of mutating the object they were called on::